packages/api/src/external/ehr/elation/command/get-patients-from-appointments.ts

```diff
@@ -0,0 +1,249 @@
+import ElationApi, { ElationEnv } from "@metriport/core/external/elation/index";
+import { executeAsynchronously } from "@metriport/core/util/concurrency";
+import { out } from "@metriport/core/util/log";
+import { capture } from "@metriport/core/util/notifications";
+import { errorToString, MetriportError } from "@metriport/shared";
+import dayjs from "dayjs";
+import duration from "dayjs/plugin/duration";
+import { getCxMappingsBySource } from "../../../../command/mapping/cx";
+import { Config } from "../../../../shared/config";
+import { EhrSources, getLookforwardTimeRange } from "../../shared";
+import { getElationClientKeyAndSecret } from "../shared";
+import { getPatientIdOrFail as singleGetPatientIdOrFail } from "./get-patient";
+
+dayjs.extend(duration);
+
+const delayBetweenPracticeBatches = dayjs.duration(0, "seconds");
+const lookforward = dayjs.duration(14, "days");
+const parallelPractices = 10;
+const parallelPatients = 2;
+
+const elationEnvironment = Config.getElationEnv();
+
+type PatientAppointment = {
+  cxId: string;
+  elationPracticeId: string;
+  elationPatientId: string;
+};
+
+export async function getPatientIdsOrFailFromAppointments(): Promise<void> {
```

Comment:
It feels strange to call the method `getSomething()` and it does some logic that's not related to returning anything. Shouldn't we call it "createPatientsFromAppointments" or something similar? (not sure why the "orFail" here)

Reply:
It *does* return something, it's just called in execute async to writes that return to `returnArray` and doesn't return it explicitly. `orFail` is that it'll error if it hits an issue, vs. returned an "undefined" appointment or something.

Reply:
Switched to process.